_posts/2021-12-27-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,346 @@
+keeps us pretty consistent with 2020s throughput.
+
+<p align="center">
+ <img align="center" width="75%" src="/assets/blog/2021-review/commits.png"/>
+</p>
+
+With the project's trajectory displayed in numbers, let's examine the top 
+features that landed in Trino this year.
+
+## Features
+
+Here's a high-level list of the most exciting features that made their way into
+Trino in 2021
+
+### Improvements to ANSI SQL support
+
+* Added support for the [`RANGE`](https://trino.io/blog/2021/03/10/introducing-new-window-features.html#new%20features)
+keyword along with an offset used within a `WINDOW` function.
+* Added [`MATCH_RECOGNIZE`](https://trino.io/blog/2021/05/19/row_pattern_matching.html)
+which is a feature added to the more recent 2016 SQL specification that allows
+for complex analysis over your data not previously available in earlier versions
+of SQL. To learn more about this feature watch [the Community Broadcast show](https://trino.io/episodes/23.html)
+about it.
+* Added support for aggregate functions in row pattern recognition context for
+`MATCH_RECOGNIZE`.
+* Added support for `TRUNCATE TABLE`.
+* Added support for `DENY` statement to in the engine.
+* Added support for [`listagg()`](https://trino.io/docs/current/functions/aggregate.html#listagg).
+* Add support for [`SET TIME ZONE`](https://trino.io/docs/current/sql/set-time-zone.html).
+* Support arbitrary queries in [`SHOW STATS`](https://trino.io/docs/current/sql/show-stats.html).
+* Added support for subquery expressions that return multiple columns. Example: 
+`SELECT x = (VALUES (1, 'a'))`.
+
+### Other query improvements
+
+* Added support for automatic query retries. This feature can be turned on by 
+setting the `retry-policy` config property or the `retry_policy` session 
+property to `QUERY` which enables query level retries.
+* Updated the behavior of `ROW` to `JSON` cast to produce `JSON` objects instead
+of `JSON` arrays.  
+
+## Performance Improvements
+
+* Improved performance of querying Parquet data for files containing column indexes.
+* Improved performance when reading dictionary-encoded Parquet files.
+* Improved performance of queries using [`rank()`](https://trino.io/docs/current/functions/window.html#rank) `WINDOW` function.
+* Improved performance of [`sum()`](https://trino.io/docs/current/functions/aggregate.html#sum)
+and [`avg()`](https://trino.io/docs/current/functions/aggregate.html#avg) for 
+decimal types.
+* Improved performance of `GROUP BY` with single grouping column.
+* Improved performance of decimal aggregations.
+* Improved performance when evaluating the WHERE and SELECT clause.
+* Improved performance when computing the product of decimal values with 
+precision larger than 19.
+* Improved performance of queries that process row or array data.
+* Reduced memory usage for processing `JOIN` clauses.
+* Improved performance of queries that contain a `DISTINCT` clause. 
+
+## Security
```

Comment:
Same as performance.. have an entry sentence and then bullet list without repeated wording